### ISSUE TITLE
[ONNX Convert] Error when input to nn.AdaptiveAvgPool2d size is variable

### ISSUE THREAD

**Issue by cengyi22** [pytorch/pytorch#147720]
### 🐛 Describe the bug

We are working on converting a trained model with nn.AdaptiveAvgPool2d to onnx, but we encountered an error where the onnx conversion is not performed when the input of nn.AdaptiveAvgPool2d is variable.

class spatial_strip_att2(nn.Module): 
    def __init__(self, dim, kernel=3, dilation=1, group=2, H=False) -> None: 
        super().__init__()
        self.k = kernel 
        pad = dilation*(kernel-1) // 2 
        self.kernel = (1, kernel) if H else (kernel, 1)
        self.padding = (kernel//2, 1) if H else (1, kernel//2) 
        self.dilation = dilation # 커널의 팽창 정도
        self.group = group # 그룹화된 convolution 수
        self.pad = nn.ReflectionPad2d((pad, pad, 0, 0)) if H else nn.ReflectionPad2d((0, 0, pad, pad)) 
        self.conv = nn.Conv2d(dim, group*kernel, kernel_size=1, stride=1, bias=False) 
        self.ap = nn.AdaptiveAvgPool2d((1, 1)) 
        self.filter_act = nn.Tanh() 
        self.inside_all = nn.Parameter(torch.zeros(dim,1,1), requires_grad=True) 
        self.lamb_l = nn.Parameter(torch.zeros(dim), requires_grad=True) 
        self.lamb_h = nn.Parameter(torch.zeros(dim), requires_grad=True)
        gap_kernel = (None,1) if H else (1, None)  
        self.gap = nn.AdaptiveAvgPool2d(gap_kernel) # Note: the bug is here, we should also pass if the gap_kernel is of variable size.

Any guidance on how to work around this, improve the code, etc. would be greatly appreciated. 

### Versions

Collecting environment information...
PyTorch version: 2.3.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro (10.0.22631 64비트)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22631-SP0
Is CUDA available: True
CUDA runtime version: 12.1.66
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4060 Laptop GPU
Nvidia driver version: 561.19
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Name: AMD Ryzen 7 8845HS w/ Radeon 780M Graphics
Manufacturer: AuthenticAMD
Family: 107
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3801
MaxClockSpeed: 3801
L2CacheSize: 8192
L2CacheSpeed: None
Revision: 29954

Versions of relevant libraries:
[pip3] numpy==1.26.3
[pip3] onnx==1.16.1
[pip3] onnxruntime-gpu==1.18.1
[pip3] onnxscript==0.1.0.dev20240708
[pip3] pytorch-lightning==2.3.0
[pip3] torch==2.3.1+cu121
[pip3] torchaudio==2.3.1+cu121
[pip3] torchmetrics==1.4.0.post0
[pip3] torchvision==0.18.1+cu121
[conda] Could not collect

**Comment by justinchuby** [comment]
Please test with `torch.onnx.export(..., dynamo=True, report=True)` using the latest torch-nightly. Attach the generated report if there is an error. Thanks!
